controllers/argocdagent/deployment.go

```diff
@@ -0,0 +1,476 @@
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalPprofPort,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalPprofPort,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalLogFormat,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalLogFormat,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalResourceProxyTLSCertPath,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalResourceProxyTLSCertPath,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalResourceProxyTLSKeyPath,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalResourceProxyTLSKeyPath,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalResourceProxyTLSCAPath,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalResourceProxyTLSCAPath,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}, {
+			Name: EnvArgoCDPrincipalEnableWebSocket,
+			ValueFrom: &corev1.EnvVarSource{
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key:                  PrincipalEnableWebSocket,
+					LocalObjectReference: ref,
+					Optional:             boolPtr(true),
+				},
+			},
+		}}
+
+	return env
+}
+
+func buildSelector(cr *argoproj.ArgoCD) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: buildLabelsForAgentPrincipal(cr.Name),
+	}
+}
+
+func buildSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		AllowPrivilegeEscalation: boolPtr(false),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{
+				"ALL",
+			},
+		},
+		ReadOnlyRootFilesystem: boolPtr(true),
+		RunAsNonRoot:           boolPtr(true),
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: "RuntimeDefault",
+		},
+	}
+}
+
+func buildPorts() []corev1.ContainerPort {
+	return []corev1.ContainerPort{
+		{
+			ContainerPort: 8443,
+			Name:          "principal",
+		}, {
+			ContainerPort: 8000,
+			Name:          "metrics",
+		},
+	}
+}
+
+func buildArgs() []string {
+	args := make([]string, 0)
+	args = append(args, "/usr/local/bin/argocd-agent-principal")
+	return args
+}
+
+func buildPrincipalImage() string {
+	image := os.Getenv("ARGOCD_AGENT_PRINCIPAL_IMAGE")
+	if image == "" {
+		image = "quay.io/openshift-gitops/argocd-agent-principal:v1"
```

Comment:
`openshift-gitops` -> `argoproj-labs`, since this is an `argoproj-labs`-based repo.